src/main/java/org/jenkinsci/plugins/workflow/cps/CpsBodyExecution.java

```diff
@@ -116,7 +118,16 @@
 
         try {
             // TODO: handle arguments to closure
-            Object x = params.body.getBody(currentThread).call();
+            Object x;
+            try {
+                x = params.body.getBody(currentThread).call();
+            } catch (InvokerInvocationException e) {
```

Comment:
There is no new exception thrown here. It's just that I discovered `CacheMethod` interferes with the exception thrown from the closure when it's a subtype of `Error`